src/adapter/graphql/tests/tests/test_gql_dataset_flow_configs.rs

```diff
@@ -0,0 +1,730 @@
+        &create_derived_result.dataset_handle.id,
+        "EXECUTE_QUERY",
+        false,
+        (30, "MINUTES"),
+        100,
+    );
+
+    let res = schema
+        .execute(
+            async_graphql::Request::new(mutation_code.clone())
+                .data(harness.catalog_authorized.clone()),
+        )
+        .await;
+
+    assert!(res.is_ok(), "{:?}", res);
+    assert_eq!(
+        res.data,
+        value!({
+            "datasets": {
+                "byId": {
+                    "flowConfigs": {
+                        "setConfigBatching": {
+                            "__typename": "FlowConfiguration",
+                            "paused": false,
+                            "schedule": Value::Null,
+                            "batching": {
+                                "__typename": "FlowConfigurationBatching",
+                                "throttlingPeriod": {
+                                    "every": 30,
+                                    "unit": "MINUTES"
+                                },
+                                "minimalDataBatch": 100
+                            }
+                        }
+                    }
+                }
+            }
+        })
+    );
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_log::test(tokio::test)]
+async fn test_incorrect_dataset_kinds_for_flow_type() {
+    let harness = FlowConfigHarness::new();
+
+    let create_root_result = harness.create_root_dataset().await;
+    let create_derived_result = harness.create_derived_dataset().await;
+
+    ////
+
+    let mutation_code = FlowConfigHarness::set_config_batching_mutation(
+        &create_root_result.dataset_handle.id,
+        "EXECUTE_QUERY",
+        false,
+        (30, "MINUTES"),
+        100,
+    );
+
+    let schema = kamu_adapter_graphql::schema_quiet();
+
+    let res = schema
+        .execute(
+            async_graphql::Request::new(mutation_code.clone())
+                .data(harness.catalog_authorized.clone()),
+        )
+        .await;
+
+    assert!(res.is_err());
+    assert_eq!(
+        res.errors
+            .into_iter()
+            .map(|e| e.message)
+            .collect::<Vec<_>>(),
+        vec!["Expected Derivative dataset kind".to_string()]
+    );
+
+    ////
+
+    let mutation_code = FlowConfigHarness::set_config_cron_expression_multation(
+        &create_derived_result.dataset_handle.id,
+        "INGEST",
+        false,
+        "0 */2 * * *",
+    );
+
+    let res = schema
+        .execute(
+            async_graphql::Request::new(mutation_code.clone())
+                .data(harness.catalog_authorized.clone()),
+        )
+        .await;
+
+    assert!(res.is_err());
+    assert_eq!(
+        res.errors
+            .into_iter()
+            .map(|e| e.message)
+            .collect::<Vec<_>>(),
+        vec!["Expected Root dataset kind".to_string()]
```

Comment:
Do we want to represent these errors as an enum result type, or you'd rather keep them as "internal errors" for now?

Reply:
Implemented similarly to `RenameResult`